packages/dataclass-editor/src/master/useKeyDown.ts

```diff
@@ -0,0 +1,35 @@
+import { useCallback, useEffect, useLayoutEffect, useRef } from 'react';
+
+export const useKeyDown = (key: string, callback: (event: KeyboardEvent) => void) => {
+  const callbackRef = useRef(callback);
+  useLayoutEffect(() => {
+    callbackRef.current = callback;
+  });
+
+  const handleKeyPress = useCallback(
+    (event: KeyboardEvent) => {
+      if (event.key === key) {
+        callbackRef.current(event);
+      }
+    },
+    [key]
+  );
+
+  useEffect(() => {
+    document.addEventListener('keydown', handleKeyPress);
+    return () => document.removeEventListener('keydown', handleKeyPress);
+  });
+};
+
+export const useShortcut = (key: string, callback: () => void) => {
+  const callbackRef = useRef(callback);
+  useLayoutEffect(() => {
+    callbackRef.current = callback;
+  });
+
+  useKeyDown(key, (event: KeyboardEvent) => {
+    if (event.ctrlKey && event.altKey) {
+      callbackRef.current();
+    }
+  });
```

Comment:
maybe we need to hold the callback as useCallback instead of directly add it here as param.